lib/DXIL/DxilResourceProperties.cpp

```diff
@@ -59,6 +59,17 @@ DXIL::ComponentType DxilResourceProperties::getCompType() const {
   return static_cast<DXIL::ComponentType>(Typed.CompType);
 }
 
+unsigned DxilResourceProperties::getElementStride() const {
+  switch (static_cast<DXIL::ResourceKind>(Basic.ResourceKind)) {
+  default:
+    return 4;
```

Comment:
It looks like for typed kinds, you can use the Typed member that contains the component number and the type. Something like: `Typed.CompComp*hlsl::CompType(static_cast<hlsl::CompType>(Typed)).GetSizeInBits()/8)`

Reply:
Agree. We could use `DxilResourceProperties::getCompType()`. It would also take care of cases where component types are not 4 bytes in length (e.g. for 16-bit types).

Reply:
Done.